src/libraries/Common/src/System/Net/Security/Unix/SafeDeleteNegoContext.cs

```diff
@@ -42,14 +43,17 @@ public SafeGssContextHandle GssContext
         }
 
         public SafeDeleteNegoContext(SafeFreeNegoCredentials credential)
-            : base(credential)
+            : base(IntPtr.Zero)
         {
             Debug.Assert((null != credential), "Null credential in SafeDeleteNegoContext");
+            _credential = credential;
```

Comment:
This should be done only after the call to DangerousAddRef.  Otherwise, if the DangerousAddRef throws, the finalizer will still try to call DangerousRelease on the _credential.

Reply:
make sense. updated.